tests/message/trailer.c

```diff
@@ -0,0 +1,179 @@
+#include "clar_libgit2.h"
+#include "message.h"
+
+struct trailer {
+	const char *key;
+	const char *value;
+};
+
+struct cb_state {
+	struct trailer *trailer;
+};
+
+static int trailer_cb(const char *key, const char *value, void *st_)
+{
+	struct cb_state *st = st_;
+
+	cl_assert_equal_s(st->trailer->key, key);
+	cl_assert_equal_s(st->trailer->value, value);
+
+	st->trailer++;
+
+	return 0;
+}
+
+static void assert_trailers(const char *message, struct trailer *trailers)
+{
+	struct cb_state st = { trailers };
+
+	int rc = git_message_trailers(message, trailer_cb, &st);
+
+	cl_assert_equal_s(NULL, st.trailer->key);
+	cl_assert_equal_s(NULL, st.trailer->value);
+
+	cl_assert_equal_i(0, rc);
+}
+
+void test_message_trailer__simple(void)
+{
+	assert_trailers(
+		"Message\n"
+		"\n"
+		"Signed-off-by: foo@bar.com\n"
+		"Signed-off-by: someone@else.com\n"
+	,
+		(struct trailer[]){
+			{ "Signed-off-by", "foo@bar.com" },
+			{ "Signed-off-by", "someone@else.com" },
+			{ NULL, NULL },
+		}
+	);
```

Comment:
I think this would be a bit more readable if you didn't declare the arrays as part of the macro but instead as variables first and then pass them into the assert_trailers. This would also get rid of the cast.

Reply:
Btw: this also causes MSVC 2010 builds to fail.